includes/class-newspack-listings-products.php

```diff
@@ -231,14 +351,1426 @@ public static function get_products() {
+					'post_status' => 'publish',
+					'post_type'   => array_values( Core::NEWSPACK_LISTINGS_POST_TYPES ),
+				]
+			);
+
+			foreach ( $associated_listings as $listing ) {
+				wp_update_post(
+					[
+						'ID'          => $listing->ID,
+						'post_status' => 'draft',
+					]
+				);
+			}
+		}
+	}
+
+	/**
+	 * Get the base URL of the current page or the My Account page, stripped of query args.
+	 *
+	 * @return string Cleaned URL.
+	 */
+	public static function get_base_url() {
+		return isset( $_SERVER['REQUEST_URI'] ) ? site_url( strtok( sanitize_text_field( $_SERVER['REQUEST_URI'] ), '?' ) ) : get_permalink( get_option( 'woocommerce_myaccount_page_id' ) );
+	}
+
+	/**
+	 * Active premium subscriptions grant customers the ability to create up to 10 Marketplace or Event listings for free.
+	 * Show controls to create and manage these listings in the Subscription account page.
+	 *
+	 * @param WC_Subscription $subscription Subscription object for the subscription whose status has changed.
+	 */
+	public static function listing_subscription_manage_premium_listings( $subscription ) {
+		$subscription_id  = $subscription->get_id();
+		$is_active        = 'active' === $subscription->get_status();
+		$is_premium       = $is_active && get_post_meta( $subscription_id, self::SUBSCRIPTION_META_KEYS['is_premium'], true );
+		$customer_id      = $subscription->get_user_id();
+		$premium_listings = get_posts(
+			[
+				'meta_key'       => self::POST_META_KEYS['listing_subscription'],
+				'meta_value'     => $subscription->get_id(), // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'post_status'    => [ 'draft', 'future', 'pending', 'private', 'publish' ],
+				'post_type'      => [
+					Core::NEWSPACK_LISTINGS_POST_TYPES['event'],
+					Core::NEWSPACK_LISTINGS_POST_TYPES['marketplace'],
+				],
+				'posts_per_page' => self::TOTAL_FREE_LISTINGS, // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+			]
+		);
+
+		// If not a premium subscription and there are no previously created premium listings, no need to show the UI.
+		if ( ! $is_premium && 0 === count( $premium_listings ) ) {
+			return;
+		}
+
+		$remaining       = self::TOTAL_FREE_LISTINGS - count( $premium_listings );
+		$base_url        = self::get_base_url();
+		$marketplace_url = wp_nonce_url(
+			add_query_arg(
+				[
+					'customer_id'     => $customer_id,
+					'subscription_id' => $subscription_id,
+					'create'          => 'marketplace',
+				],
+				$base_url
+			),
+			self::CREATE_LISTING_NONCE
+		);
+		$event_url       = wp_nonce_url(
+			add_query_arg(
+				[
+					'customer_id'     => $customer_id,
+					'subscription_id' => $subscription_id,
+					'create'          => 'event',
+				],
+				$base_url
+			),
+			self::CREATE_LISTING_NONCE
+		);
+
+		?>
+			<h2><?php echo esc_html( __( 'Premium subscription listings', 'newspack-listings' ) ); ?></h2>
```

Comment:
The UI of this section is hard to understand. Here's how it looks after buying a non-premium subscription:

<img width="869" alt="image" src="https://user-images.githubusercontent.com/7383192/144249430-50124efa-3e8a-44f6-a07d-64d0c8302f06.png">

The non-premium subscription should be in a different section. When seeing this UI for the first time, I thought I'm supposed to buy additional listings on top of a subscription.

Reply:
This is actually a bug due to a mismatch between two variable names...a param wasn't being passed to the order handler, so subscriptions were generating Marketplace listings instead of the intended Place listing.

This should be fixed in 88e9838. Now, after purchasing a non-premium subscription, you shouldn't see the "Premium subscription listings" section at all, and the listing associated with the subscription order should be a Place listing.